src/praisonai-agents/praisonaiagents/workflows/workflows.py

```
@@ -2270,25 +2301,102 @@ def _execute_single_step_internal(
+        """
+        max_retries = getattr(step, 'max_retries', 3)
+        retry_count = 0
+        validation_feedback = None
+        output = None
+        step_error = None
+
+        while retry_count <= max_retries:
+            step_error = None
+            try:
+                output = run_body(validation_feedback)
+            except Exception as e:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)